### PR TITLE
withValue() -> with { value }

### PR DESCRIPTION
**Pull request by Jolanrensen**
Discussed in https://github.com/Kotlin/dataframe/issues/253#issuecomment-1587226245

Since we have no plans atm to implement single-value columns, I think it might be best to remove `withValue()` from the `Update` function (and the docs of `Convert`).

This PR replaces usage of it across the library and examples and provides a helpful deprecation annotation.

### CODE REVIEW

**Review by zaleslaw — APPROVED** [review]

File: examples/notebooks/titanic/Titanic.ipynb [hunk 1]
```diff
@@ -1111,7 +1111,7 @@
     "val df1 = df\n",
     "    // imputing\n",
     "    .fillNulls { sibsp and parch and age and fare }.perCol { mean() }\n",
-    "    .fillNulls { sex }.withValue(\"female\")\n",
+    "    .fillNulls { sex }.with { \"female\" }\n",
```
Comment (koperagen) [inline examples/notebooks/titanic/Titanic.ipynb:1114]: Is it updated on datalore?

**Review by koperagen — APPROVED** [review]

File: examples/notebooks/titanic/Titanic.ipynb (hunk 1 above)
Comment (Jolanrensen) [reply examples/notebooks/titanic/Titanic.ipynb:1114]: no, datalore runs the release version of DF, updating it now would break it

Comment (koperagen) [reply examples/notebooks/titanic/Titanic.ipynb:1114]: Hm, probably not in this case, `with` is available in 0.10.1

Comment (Jolanrensen) [reply examples/notebooks/titanic/Titanic.ipynb:1114]: oh of course XD, you're right

Comment (Jolanrensen) [reply examples/notebooks/titanic/Titanic.ipynb:1114]: done!

**Pull request merged**